### PR TITLE
Add Unison language

### PR DESCRIPTION
**Pull request by ereslibre**

### CODE REVIEW

File: src/modules/languages/unison.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.languages.unison;
+in
+{
+  options.languages.unison = {
+    enable = lib.mkEnableOption "Enable tools for Unison development.";
+
+    package = lib.mkOptionn {
```
Comment (domenkozar) [inline src/modules/languages/unison.nix:10]:
```suggestion
    package = lib.mkOption {
```

Comment (ereslibre) [reply src/modules/languages/unison.nix]: Ooops, I have no idea how that happened!

**Comment by domenkozar** [comment]
It's now complaining that `unison-code-manager` is marked as unfree, do you know if that's correct?

**Comment by domenkozar** [comment]
See https://github.com/cachix/devenv/actions/runs/5377840454/jobs/9756883028#step:8:442

**Comment by ereslibre** [comment]
> so it seems like we need to add back arm/linux support.

I don't think it ever had it. The binaries they provide with their automation for Linux is x86_64 only.

From: https://www.unison-lang.org/learn/quickstart/#installation-options, "The current alpha release is for Mac OS X, 64-bit Linux, and Windows users!"

**Pull request merged**